### ISSUE TITLE
🐞 F1AdaptiveThreshold does not work as expected if anomalous examples are missing

### ISSUE THREAD

**Issue by nicoschena** [open-edge-platform/anomalib#3357]
### Describe the bug

in metrics/threshold/f1_adaptive_threshold.py
In absence of anomalous example, the compute method of **_F1AdaptiveThreshold** is supposed to return the **maximum anomaly scores** as written in the message:

            msg = (
                "The validation set does not contain any anomalous images. As a "
                "result, the adaptive threshold will take the value of the "
                "highest anomaly score observed in the normal validation images, "
                "which may lead to poor predictions. For a more reliable "
                "adaptive threshold computation, please add some anomalous "
                "images to the validation set."
            )

**Instead it returns the minimum anomaly score, making all prediction anomalous.**

The:
_return thresholds[torch.argmax(f1_score)]_

**returns the minimum threshold since torch.argmax(f1_score) returns the first element if all elements are zero.**

### Dataset

N/A

### Model

N/A

### Steps to reproduce the behavior

1. Install anomalib 2.2.0 from python pypi
2. Train an anomaly detection model without anomalous example in val/test set (I have tried PatchCore and Dinomaly)
3. Perform prediction on good test examples

### OS information

Not necessary

### Expected behavior

As the message states, the maximum anomaly value seen during validation (when no anomalous example are given) should be returned.

### Pip/GitHub

pip

### What version/branch did you use?

2.2.0

### Configuration YAML

```yaml
-
```

### Logs

```shell
-
```

### Code of Conduct

- [x] I agree to follow this project's Code of Conduct
